src/shared/components/oncoprint/ResultsViewOncoprint.tsx

```diff
@@ -191,6 +196,53 @@ export default class ResultsViewOncoprint extends React.Component<IResultsViewOn
             }
         );
 
+        // update URL parameters according to sort direction UI events
+        // and DO NOT trigger a page refresh
+        this.urlParamsReaction = reaction(
+            ()=>[
+                this.sortMode,
+                this.sortByMutationType,
+                this.sortByDrivers
+            ],
+            ()=>{
+                const newParams = Object.assign({}, getBrowserWindow().globalStores.routing.location.query);
+                if (this.sortMode) {
+                    const mode = this.sortMode.type;
+                    let sortby_param = "";
+                    switch (mode) {
+                        case "alphabetical":               // sort by sample or patient id (a.k.a. alphabetical)
+                            sortby_param = "case_id";
+                            break;
+                        case "caseList":                   // sort by order of appearance in case list (when selected)
+                            sortby_param = "case_list";
+                            break;
+                        case "data":                        // sort by data
+                            sortby_param = "data";
+                            if (this.sortByMutationType) {
+                                sortby_param += "_mut";            // sort by data + mutation type
+                            }
+                            if (this.sortByDrivers) {       // sort by  data + driver/passenger
+                                sortby_param += "_dri";
+                            }
+                            break;
+                        default:
+                            sortby_param = "";
+                            break;
+                    }
+                    newParams[ONCOPRINT_SORTBY_URL_PARAM] = sortby_param;
+                } else {
+                    delete newParams[ONCOPRINT_SORTBY_URL_PARAM];
+                }
+
+                // update URL without page refresh
+                const {basename, pathname} = getBrowserWindow().globalStores.routing.location;
```

Comment:
this block will be redundant once you move your changes into the other reaction, but just FYI you can just do it like this:

`getBrowserWindow().globalStores.routing.updateRoute(newParams, undefined, true, true);` as we do on line 195

Reply:
Again, I put these in separate reaction because I do **not** want to trigger a page update and AFAIKS `getBrowserWindow().globalStores.routing.updateRoute(newParams, undefined, true, true);` does trigger a refresh.

If I am mistaken I woul dbe more than happy to move, but first plz confirm that you understand that the URL action for the sortBy param differs from that of  columnMode, heatmapTrackGroupsUrlParam and clinicalTracksUrlParam.

Reply:
@pvannierop can you clarify what you mean by "refresh"?   if you mean reload of actual HTML page, it should not do that.   it may trigger re-render of whole oncoprint.  the reason is that it changes what we think of as the "query."   there is a complication in all this b/c for large queries we cannot store all params in url.  we have to save it to session.    should these params be considered part of that session?  would a user want these settings to be reproduced when sharing the query with another user?   ultimately, we may need to decouple what we think of as the session from some settings which are also stored in the url.  complicated, i know!  perhaps we should have a meeting to go over this.

Reply:
yes the existing reaction definitely does not cause a refresh, please clarify

Reply:
@pvannierop still confused what you mean - neither reaction causes a refresh